Core/src/org/sleuthkit/autopsy/modules/ileappanalyzer/ILeappAnalyzerIngestModule.java

```diff
@@ -0,0 +1,263 @@
+
+    private static final Logger logger = Logger.getLogger(ILeappAnalyzerIngestModule.class.getName());
+    private static final String MODULE_NAME = ILeappAnalyzerModuleFactory.getModuleName();
+
+    private static final String ILEAPP = "iLeapp"; //NON-NLS
+    private static final String ILEAPP_EXECUTABLE = "ileapp.exe";//NON-NLS
+
+    private File iLeappExecutable;
+
+    private IngestJobContext context;
+    
+    private ILeappFileProcessor iLeappFileProcessor;
+
+    ILeappAnalyzerIngestModule() {
+        // This constructor is intentionally empty. Nothing special is needed here.     
+    }
+
+    @NbBundle.Messages({
+        "ILeappAnalyzerIngestModule.executable.not.found=iLeapp Executable Not Found.",
+        "ILeappAnalyzerIngestModule.requires.windows=iLeapp module requires windows.",
+        "ILeappAnalyzerIngestModule.error.ileapp.file.processor.init=Failure to initialize ILeappProcessFile"})
+    @Override
+    public void startUp(IngestJobContext context) throws IngestModuleException {
+        this.context = context;
+
+        if (false == PlatformUtil.isWindowsOS()) {
+            throw new IngestModuleException(Bundle.ILeappAnalyzerIngestModule_requires_windows());
+        }
+
+        try {
+            iLeappFileProcessor = new ILeappFileProcessor();
+        } catch (IOException | IngestModuleException ex) {
+            throw new IngestModuleException(Bundle.ILeappAnalyzerIngestModule_error_ileapp_file_processor_init(), ex);
+        }
+
+        try {
+            iLeappExecutable = locateExecutable(ILEAPP_EXECUTABLE);
+        } catch (FileNotFoundException exception) {
+            logger.log(Level.WARNING, "iLeapp executable not found.", exception); //NON-NLS
+            throw new IngestModuleException(Bundle.ILeappAnalyzerIngestModule_executable_not_found(), exception);
+        }
+
+    }
+
+    @NbBundle.Messages({
+        "ILeappAnalyzerIngestModule.error.running.iLeapp=Error running iLeapp, see log file.",
+        "ILeappAnalyzerIngestModule.error.creating.output.dir=Error creating iLeapp module output directory.",
+        "ILeappAnalyzerIngestModule.starting.iLeapp=Starting iLeapp",
+        "ILeappAnalyzerIngestModule.running.iLeapp=Running iLeapp",
+        "ILeappAnalyzerIngestModule.has.run=iLeapp",
+        "ILeappAnalyzerIngestModule.iLeapp.cancelled=iLeapp run was canceled",
+        "ILeappAnalyzerIngestModule.completed=iLeapp Processing Completed",
+        "ILeappAnalyzerIngestModule.report.name=iLeapp Html Report"})
+    @Override
+    public ProcessResult process(Content dataSource, DataSourceIngestModuleProgress statusHelper) {
+
+        statusHelper.progress(Bundle.ILeappAnalyzerIngestModule_starting_iLeapp(), 0);
+
+        List<AbstractFile> iLeappFilesToProcess = findiLeappFilesToProcess(dataSource);
+
+        statusHelper.switchToDeterminate(iLeappFilesToProcess.size());
+
+        Integer filesProcessedCount = 0;
+
+        if (!iLeappFilesToProcess.isEmpty()) {
```

Comment:
Is this check necessary? I think you can just let the for loop recognize there is nothing to process.

Reply:
It was removed and ran fine